File: Lib/Telerik/Examples/QuickStart/GridView/Spreadsheet/SpreadsheetGridRowHeaderCellElement.vb
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.GridView.Spreadsheet
	Public Class SpreadsheetGridRowHeaderCellElement
		Inherits GridRowHeaderCellElement
		#Region "Initialization"

		Public Sub New(ByVal column As GridViewColumn, ByVal row As GridRowElement)
			MyBase.New(column, row)

		End Sub

		Public Overrides Sub Initialize(ByVal column As GridViewColumn, ByVal row As GridRowElement)
			MyBase.Initialize(column, row)
		End Sub

		Protected Overrides Sub InitializeFields()
			MyBase.InitializeFields()

			MyBase.DrawFill = True
			MyBase.GradientStyle = Telerik.WinControls.GradientStyles.Linear
			MyBase.NumberOfColors = 4
			MyBase.BackColor = Color.FromArgb(231, 240, 252)
			MyBase.BackColor2 = Color.FromArgb(218, 233, 251)
			MyBase.BackColor3 = Color.FromArgb(216, 231, 251)
			MyBase.BackColor4 = Color.FromArgb(209, 226, 247)
			MyBase.DrawBorder = True
			MyBase.BorderGradientStyle = Telerik.WinControls.GradientStyles.Solid
			MyBase.BorderBoxStyle = Telerik.WinControls.BorderBoxStyle.OuterInnerBorders
			MyBase.BorderColor = Color.FromArgb(209, 225, 245)
			MyBase.BorderInnerColor = Color.White
		End Sub

		Public Overrides Sub SetContent()
			Dim rowNumber As Integer = Me.ViewTemplate.Rows.IndexOf(Me.RowInfo) + 1
			Me.Text = rowNumber.ToString()
		End Sub

		#End Region
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/Spreadsheet/SpreadsheetGridRowHeaderCellElement.cs
using System.Drawing;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.GridView.Spreadsheet
{
    public class SpreadsheetGridRowHeaderCellElement : GridRowHeaderCellElement
    {
        #region Initialization

        public SpreadsheetGridRowHeaderCellElement(GridViewColumn column, GridRowElement row)
            : base(column, row)
        {
            
        }

        public override void Initialize(GridViewColumn column, GridRowElement row)
        {
            base.Initialize(column, row);
        }

        protected override void InitializeFields()
        {
            base.InitializeFields();

            base.DrawFill = true;
            base.GradientStyle = Telerik.WinControls.GradientStyles.Linear;
            base.NumberOfColors = 4;
            base.BackColor = Color.FromArgb(231, 240, 252);
            base.BackColor2 = Color.FromArgb(218, 233, 251);
            base.BackColor3 = Color.FromArgb(216, 231, 251);
            base.BackColor4 = Color.FromArgb(209, 226, 247);
            base.DrawBorder = true;
            base.BorderGradientStyle = Telerik.WinControls.GradientStyles.Solid;
            base.BorderBoxStyle = Telerik.WinControls.BorderBoxStyle.OuterInnerBorders;
            base.BorderColor = Color.FromArgb(209, 225, 245);
            base.BorderInnerColor = Color.White;
        }

        public override void SetContent()
        {
            int rowNumber = this.ViewTemplate.Rows.IndexOf(this.RowInfo) + 1;
            this.Text = rowNumber.ToString();
        }

        #endregion
    }
}
